### ISSUE TITLE
nan in training loss

### ISSUE THREAD

**Issue by prajwalsingh** [junshengzhou/3DAttriFlow#4]
Thank you for making the code publicly available. 

I just have one small query related to training the network from scratch. After running the training code, I was getting the Nan loss. I traced it back to the initialization in SP_DecoderEigen3steps -> SubSpaceLayer:

self.U = nn.Parameter(torch.empty(n_basis, dim))
nn.init.orthogonal_(self.U)

The orthogonal constraint causes the weights to go Nan. For testing purposes, I changed it to:

self.U = nn.Parameter(torch.nn.init.xavier_uniform(torch.empty(n_basis, dim))+1e-6)

This solves the problem of Nan, but I am not sure if this will converge.

I am looking forward to your help.

**Comment by junshengzhou** [comment]
Hi,

Thanks for your interest! I didn't observe the Nan loss problem during my training, so I am not quiet sure what caused it. But I believe the small difference in parameter initialization will not affect the convergence of the network.

**Comment by prajwalsingh** [comment]
Good to know this. Currently, the training is in progress, and loss is converging.

Thank you for your quick response and help.
